src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -1566,24 +1566,26 @@ internal static int IndexOfChar(ref char searchSpace, char value, int length)
                 // Loop until either we've finished all elements or there's less than a vector's-worth remaining.
                 do
                 {
-                    equals = TNegator.NegateIfNeeded(Vector512.Equals(values, Vector512.LoadUnsafe(ref currentSearchSpace)));
-                    if (equals == Vector512<TValue>.Zero)
+                    equals = Vector512.LoadUnsafe(ref currentSearchSpace);
+
+                    if (TNegator.NegateIfNeeded(!Vector512.EqualsAny(values, equals)))
                     {
                         currentSearchSpace = ref Unsafe.Add(ref currentSearchSpace, Vector512<TValue>.Count);
                         continue;
                     }
 
-                    return ComputeFirstIndex(ref searchSpace, ref currentSearchSpace, equals);
+                    return ComputeFirstIndex(ref searchSpace, ref currentSearchSpace, TNegator.NegateIfNeeded(Vector512.Equals(values, equals)));
```

Comment:
likewise in this file. Are these checks even 1 more than previous, gives better performance?